queries.sql

```diff
@@ -66,3 +66,230 @@ SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+  9 | Charmander | 2020-02-08    |               0 | f        |        11 | pokemon
+(10 filas)
+*/
+
+ROLLBACK TO before_delete_date;
+SELECT * FROM animals;
+/*
+ id |    name    | date_of_birth | escape_attempts | neutered | weight_kg | species 
+----+------------+---------------+-----------------+----------+-----------+---------
+  1 | Agumon     | 2020-02-03    |               0 | t        |     10.23 | digimon
+  6 | Gabumon    | 2018-11-15    |               2 | t        |         8 | digimon
+  8 | Devimon    | 2017-05-12    |               5 | t        |        11 | digimon
+ 10 | Plantmon   | 2021-11-15    |               2 | t        |      -5.7 | digimon
+ 12 | Angemon    | 2005-06-12    |               1 | t        |       -45 | digimon
+ 13 | Boarmon    | 2005-06-07    |               7 | t        |      20.4 | digimon
+  7 | Pikachu    | 2021-01-07    |               1 | f        |     15.04 | pokemon
+ 11 | Squirtle   | 1993-04-02    |               3 | f        |    -12.13 | pokemon
+ 14 | Blossom    | 1998-10-13    |               3 | t        |        17 | pokemon
+  9 | Charmander | 2020-02-08    |               0 | f        |       -11 | pokemon
+(10 filas)
+*/
+
+UPDATE animals SET weight_kg=weight_kg*-1 
+WHERE weight_kg<0;
+SELECT * FROM animals;
+/*
+ id |    name    | date_of_birth | escape_attempts | neutered | weight_kg | species 
+----+------------+---------------+-----------------+----------+-----------+---------
+  1 | Agumon     | 2020-02-03    |               0 | t        |     10.23 | digimon
+  6 | Gabumon    | 2018-11-15    |               2 | t        |         8 | digimon
+  8 | Devimon    | 2017-05-12    |               5 | t        |        11 | digimon
+ 13 | Boarmon    | 2005-06-07    |               7 | t        |      20.4 | digimon
+  7 | Pikachu    | 2021-01-07    |               1 | f        |     15.04 | pokemon
+ 14 | Blossom    | 1998-10-13    |               3 | t        |        17 | pokemon
+ 10 | Plantmon   | 2021-11-15    |               2 | t        |       5.7 | digimon
+ 12 | Angemon    | 2005-06-12    |               1 | t        |        45 | digimon
+ 11 | Squirtle   | 1993-04-02    |               3 | f        |     12.13 | pokemon
+  9 | Charmander | 2020-02-08    |               0 | f        |        11 | pokemon
+*/
+COMMIT;
+
+/* ****************************************************************************************************** */
+
+/* How many animals are there? */
+SELECT COUNT(*) FROM animals;
+/*
+ count 
+-------
+    10
+(1 fila)
+*/
+
+/* How many animals have never tried to escape? */
+SELECT COUNT(*) FROM animals 
+WHERE escape_attempts=0;
+/*
+ count 
+-------
+     2
+(1 fila)
+*/
+
+/* What is the average weight of animals? */
+SELECT AVG(weight_kg) FROM animals;
+/*
+---------------------
+ 15.5500000000000000
+(1 fila)*/
+
+/* Who escapes the most, neutered or not neutered animals? */
+SELECT neutered, SUM(escape_attempts) FROM animals 
+GROUP BY neutered;
+/*
```

Comment:
I have a conflict here, The question is who scapes the most if the neutered or the not neutered animals. If I use MAX, I will get the max attempts but just one animal per group. It could be for example, that just one neutered animal tried to escape 8 times but just that one, and maybe 5 animals not neutered tried to escape 5, 3, and 7 times. In this case, the not neutered tried to escape more times than neutered animals. My solution is if you don't want that I use SUM, create an average of escapes from neutered and not neutered like this:
SELECT neutered, AVG(escape_attempts) as AVG_escape
FROM animals
GROUP BY neutered;
![Captura desde 2023-02-20 21-12-10](https://user-images.githubusercontent.com/113067895/220239632-8970d985-3522-427d-b23c-361168f7df18.png)